orchestrator/domain/base.py

```diff
@@ -1156,6 +1157,75 @@ def from_other_lifecycle(
 
         return model
 
+    # TODO: this method has quite a lot in common with from_subscription. If we are happy with this approach we can
+    # extract some common functions
+    @classmethod
+    def from_other_product(
+        cls: Type[S],
+        old_instantiation: S,
+        new_product_id: Union[UUID, str],
+        new_root: Optional[tuple[str, ProductBlockModel]] = None,
+    ) -> S:
+        db_product = get_product_by_id(new_product_id)
+
+        subscription = SubscriptionTable.query.options(
+            selectinload(SubscriptionTable.instances)
+            .selectinload(SubscriptionInstanceTable.product_block)
+            .selectinload(ProductBlockTable.resource_types),
+            selectinload(SubscriptionTable.instances).selectinload(SubscriptionInstanceTable.parent_relations),
+            selectinload(SubscriptionTable.instances).selectinload(SubscriptionInstanceTable.values),
+        ).get(old_instantiation.subscription_id)
+
+        product = ProductModel(
+            name=db_product.name,
+            description=db_product.description,
+            product_id=db_product.product_id,
+            tag=db_product.tag,
+            product_type=db_product.product_type,
+            status=db_product.status,
+        )
+
+        status = SubscriptionLifecycle(subscription.status)
+
+        if not cls.__base_type__:
+            # Import here to prevent cyclic imports
+            from orchestrator.domain import SUBSCRIPTION_MODEL_REGISTRY
+
+            cls = SUBSCRIPTION_MODEL_REGISTRY.get(subscription.product.name, cls)  # type:ignore
+            cls = lookup_specialized_type(cls, status)
+        elif not issubclass(cls, lookup_specialized_type(cls, status)):
+            raise ValueError(f"{cls} is not valid for lifecycle {status}")
+
+        fixed_inputs = {fi.name: fi.value for fi in subscription.product.fixed_inputs}
```

Comment:
These retrieves the fixed_input values from the current product. If this function's purpose is to change the subscription to a new product shouldn't this retrieve the fixed_input values of the new product instead?

Reply:
Should indeed be the fixed inputs of the new product.

Reply:
Good  catch. I was a bit too fast copy/pasting coding from from_subscription. Fixed.